CRM/Core/BAO/FinancialTrxn.php

```diff
@@ -495,10 +497,14 @@ public static function getTotalPayments($contributionID, $includeRefund = FALSE)
       INNER JOIN civicrm_entity_financial_trxn eft ON (eft.financial_trxn_id = ft.id AND eft.entity_table = 'civicrm_contribution')
       WHERE eft.entity_id = %1 AND ft.is_payment = 1 AND ft.status_id IN (%2) ";
 
-    return CRM_Core_DAO::singleValueQuery($sql, [
+    $totalAmount = CRM_Core_DAO::singleValueQuery($sql, [
       1 => [$contributionID, 'Integer'],
       2 => [implode(',', $statusIDs), 'CommaSeparatedIntegers'],
     ]);
+    if (!$totalAmount) {
+      $totalAmount = 0;
+    }
+    return $totalAmount;
```

Comment:
I think you need to cast to float here to be sure

Reply:
@eileenmcnaughton Yes. fixed.